Add test for setting min worker threads in thread pool to 0

kouvel opened the pull request:
Related to https://github.com/dotnet/coreclr/issues/14239
Depends on https://github.com/dotnet/coreclr/pull/14864

stephentoub commented on src/System.Threading.ThreadPool/tests/ThreadPoolTests.cs:
```diff
@@ -195,6 +195,47 @@ private static void VerifyMaxThreads(int expectedMaxw, int expectedMaxc)
+                    }
+                };
+                ThreadPool.QueueUserWorkItem(callback);
+                done.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds);
```
What are you trying to test here?

kouvel replied:
I'm testing that it makes progress on work items even though the min thread count is set to 0 since that's an allowed value. There was an assertion failure in this case that is fixed as well, so a bit of coverage for that as well.

stephentoub replied:
I see.  Ok.

kouvel commented:
@dotnet-bot test Windows x86 Release Build

kouvel commented:
Flagging as no-merge until dependency is reflected in CoreCLR package used by CoreFX. I guess CoreFX tests only run on release CoreCLR, otherwise there would be a test failure.

kouvel commented:
Rebased on top of https://github.com/dotnet/corefx/pull/25144. That would be merged separately, ignore first commit. Added verification that was removed in that PR.

kouvel commented:
Rebased after #25144 was merged

stephentoub commented on src/System.Threading.ThreadPool/tests/ThreadPoolTests.cs:
```diff
@@ -194,6 +195,48 @@ private static void VerifyMaxThreads(int expectedMaxw, int expectedMaxc)
+            {
+                Assert.True(ThreadPool.SetMaxThreads(maxw, maxc));
+                Assert.True(ThreadPool.SetMinThreads(minw, minc));
+            }
```
Just curious why you chose to write a local function for this rather than just putting these two lines into the finally block?

kouvel replied:
I'll change it

stephentoub commented:
@dotnet-bot test this please

mmitche commented:
@dotnet-bot test this please (restarted for plugin update)

kouvel commented:
@dotnet-bot test this please
